cpu/stm32/cpu_init.c

```diff
@@ -152,6 +153,135 @@ static void _gpio_init_ain(void)
 }
 #endif
 
+/**
+ * @brief get the value of a register in a glitch resistant fashion
+ *
+ * This very teniously avoids optimization, even optimized it's better than
+ * nothing but periodic review should establish that it doesn't get optimized.
+ */
+__attribute__((always_inline))
+static inline uint32_t _multi_read_reg32(volatile uint32_t *addr)
+{
+    uint32_t value = *addr;
+// cppcheck-suppress duplicateExpression
+// cppcheck-suppress knownConditionTrueFalse
+    if (*addr != value || *addr != value) {
+        /* (reason: volatile pointer forces multiple reads for glitch resistance,
+         glitch may force different value) */
+        while(1);
```

Comment:
shouldn't this also call `pm_set(0)` - or better yet
```suggestion
        *glitch = true;
```

with 

```C
static inline uint32_t _multi_read_reg32(volatile uint32_t *addr, bool *glitch)
```